modules/creating-manifest-file-multiple-interfaces-br-ex.adoc

```diff
@@ -0,0 +1,177 @@
+====
++
+.. Define a secondary interface in a machine config manifest file:
++
+[source,yaml]
+----
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: master
+  name: 11-master-bond0
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    storage:
+      files:
+      - contents:
+          source: data:text/plain;base64,<encoded_base64_string> <1>
+        path: /etc/NetworkManager/system-connections/bond0.nmconnection
+        filesystem: root
+        mode: 0600
+      - contents:
+# ...
+----
+<1> The Base64 encoded string that you have generated in the previous step.
++
+.. Create a connection profile for the bonded secondary interface and save the file in the `/etc/NetworkManager/system-connections/secondary-iface.nmconnection` path. For example, create the `bond1.nmconnection` file in your local directory with the following content:
++
+[source,ini]
+----
+[connection]
+id=eno2
+type=ethernet
+interface-name=eno2
+master=bond0
+slave-type=bond
+autoconnect=true
+autoconnect-priority=20
+
+[ethernet]
+mtu=9000
+----
+
+. Create an Open vSwitch (OVS) bridge, `br-ex1`, on nodes by using OVN-Kubernetes network plugin to create an `extra_bridge` file`. Ensure that you save the file in the `/etc/ovnk/extra_bridge` path of the host. The file must state the interface name that supports the additional bridge and not the default interface that supports `br-ex`, which holds the primary IP address of the node. 
++
+.Example configuration for the `extra_bridge` file, `/etc/ovnk/extra_bridge`, that references a secondary interface that is attached to a VLAN
+[source,text]
+----
+ens4f1.1001
+----
+
+. Prepare the environment variables. Replace `<machine_role>` with the node role, such as `worker`, and replace `<interface_name>` with the name of your additional `br-ex` bridge name.
++
+[source,terminal]
+----
+$ export ROLE=<machine_role>
+----
++
+[source,terminal]
+----
+$ export INTERFACE=<interface_name>
+----
+
+. Create a machine config manifest file for configuring the network plugin by entering the following command in your terminal:
++
+[source,terminal]
+----
+$ oc create -f sec-bridge-cni.yaml
+----
+
+. Edit the machine config YAML file to have a configuration analogous to the following example:
++
+[source,yaml]
+----
+# ...
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: ${ROLE}
+  name: 12-${ROLE}-sec-bridge-cni
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    storage:
+      files:
+        - path: /etc/ovnk/extra_bridge
+          mode: 0420
+          overwrite: true
+          contents:
+            source: data:text/plain;charset=utf-8,${INTERFACE}
+          filesystem: root
+# ...
+----
+
+// Map nodes in your cluster to use the additional bridge by configuring the `/var/lib/ovnk/iface_default_hint` file.
```

Comment:
The `/varl/lib/ovnk/iface_default_hint` file will only be referenced by configure-ovs.sh as a process to define `br-ex` (so it accepts one value). The contents of this file should only be: `bond0` (assuming bond0 exists and we want to use bond1 for our br-ex1 connection). Objectively this file should be used to ensure that the default configuration comes up on the desired iface and not the new one that has been configured. (optional). I've reached out to the engineering team to see if it will accept a second value but my understanding at present is this is a direct variable inference --> https://github.com/trozet/machine-config-operator/blob/1ae564e4bc061753c5e240f8f706cf97bad18f30/templates/common/_base/files/configure-ovs-network.yaml#L570-L585

Reply:
Update: `br-ex` can be overridden with `/var/lib/ovnk/iface_default_hint` and `br-ex1` is overridden/defined with `etc/ovnk/extra_bridge`. That's why each of these files have exactly one argument which is the name of the iface to be selected. between the two of them you can point your config however you want.

Reply:
:heavy_check_mark: